### DISCUSSION TITLE
Street Names in Irish Language

### DISCUSSION THREAD

**Discussion by jonnymccullagh** [openmaptiles/openmaptiles#1569] [General]
Hi, I am interested in displaying maps showing the Irish (`name:ga`) language values as the default. I had a quick play with the MapTiler switching the language to Irish but that only showed the towns and cities translated, and not street names. I tried the Welsh version and it seemed to have street names just as I would have expected. So I'd love it if someone could point me in the right direction for building the map tiles myself with the street names included.

**Comment by TomPohys** [comment]
Hi, could you please provide some locations, links, or examples where the switch language not working?

Thanks

**Reply by 1ec5** [reply]
openmaptiles/planetiler-openmaptiles#58 describes a problem along these lines in more detail. (Unsure if it also affects openmaptiles/openmaptiles.)

**Reply by jonnymccullagh** [reply]
In the MapTiler editor I had zoomed in to this URL:
https://cloud.maptiler.com/maps/editor?map=streets-v2#17.03/54.181256/-6.324949
Attached images show the view on Maptiler and the relevant node on OSM:
<img width="519" alt="image" src="https://github.com/openmaptiles/openmaptiles/assets/5209315/0e24cde1-5b1f-486d-bd85-0a633ca0acbb">

<img width="485" alt="image" src="https://github.com/openmaptiles/openmaptiles/assets/5209315/9bae4a3c-9939-49d3-931c-6999024b7501">

I thought I could maybe build the tiles myself providing a switch to also include the street names.

**Reply by ZeLonewolf** [reply]
Is the maptiler planet using the latest openmaptiles?  Because in the issue ticket that @1ec5 linked, omt-tools was producing the language versions.

**Reply by jonnymccullagh** [reply]
Go easy on me here guys ;-) I'm a noob. I don't know omt-tools is, who Minh is and I'm guessing that maptiler planet is their cloud hosted tile service. But if this is solved in the repo then should I just have a go at building it ?

**Reply by ZeLonewolf** [reply]
It seems to me that MapTiler Cloud may be missing something that works in current openmaptiles.

**Reply by jonnymccullagh** [reply]
ok thanks, I'll try building it myself :crossed_fingers:

**Reply by jonnymccullagh** [reply]
I have now built vector tiles for Ireland using OpenMapTiles amending the `mapping.yaml` file before generating:
```
    - name: name
      key: name
      type: string
    - name: name_en
      key: name:en
      type: string
    - name: name_de
      key: name:de
      type: string
```
for:
```
    - name: name
      key: name
      type: string
    - name: name_en
      key: name:en
      type: string
    - name: name_ga
      key: name:ga
      type: string
``` 
I am serving the tiles using TileServer-GL and rendering in a web page with MapLibre-GL. This works fine for English, however I am still unable to view names in the Irish language. 

I've tried:
- setting my browser language
- calling the styles.json with a language parameter(`&language=ga`)
- attempted to use the button examples for language switching

So I am looking for some guidance as to where I might be going wrong and in which component. 
- Should I be calling a different styles URL to get a different language ? 
- Do I try to tell TileServer-GL to serve in a specific language ? 
- Can I confirm that my generated tiles have the language items for ga ?

**Reply by 1ec5** [reply]
> Should I be calling a different styles URL to get a different language ?
> Do I try to tell TileServer-GL to serve in a specific language ?

I don’t think TileServer GL supports dynamically changing the `name` property dynamically based on a URL parameter, though that would be a nice feature.

> Can I confirm that my generated tiles have the language items for ga ?

Go to the inspect.html page generated by TileServer GL and hover over a street; you should see `name:ga` listed among the feature’s properties. But this alone isn’t enough; the style you’re using also needs to label features based on this property instead of the usual `name` property. This is what the “Language” setting in MapTiler’s editor is probably doing behind the scenes. In the style JSON, any road-related `symbol` layer should have a `text-field` property set to a string like `{name:ga}` or an expression like `["get", "name:ga"]`.

**Reply by jonnymccullagh** [reply]
@1ec5 perfect - that got me sorted. I have now locally served a map with the street names in the language I wanted. Next step is a write-up and hosting it somewhere. Thanks everyone for your great help!

**Comment by nguyenhoaibaoduc** [comment]
I know , because with version translate of apple , Apple's translation is backwards, it doesn't recognize between English, you should use English (UK), don't use English local

**Comment by nguyenhoaibaoduc** [comment]
you should find a way to hide before you build it, because IP is for local, almost all are managed by local, you should build it in another Ip area, because of the saying, "All Map Is One" :))) ))
